tcms/core/ajax.py

```diff
@@ -233,29 +233,28 @@ def add(self):
                     for o in self.obj:
                         o.add_tag(tag)
                 except IntegrityError:
-                    return "Error when adding %s " \
+                    return False, "Error when adding %s " \
                            "Tag with this key already exists in the database." % self.tag
                 except Exception:
-                    return "Error when adding %s" % self.tag
+                    return False, "Error when adding %s" % self.tag
 
             return True, self.obj
 
         def remove(self):
             self.obj = self.obj.filter(tag__name__in=self.tag).distinct()
 
             if not self.obj:
-                return "Tags does not exist in current selected plan."
+                return False, "Tags does not exist in current selected plan."
 
             for tag_str in self.tag:
                 try:
                     tag = Tag.objects.filter(name=tag_str)[0]
-                except IndexError:
-                    return "Tag %s does not exist in current selected plan." % tag_str
-                for object in self.obj:
-                    try:
+                    for object in self.obj:
```

Comment:
It looks like you have changed indentation. Why ?

Reply:
I have merged the two try-except blocks. The for loop goes into the first try.